test/distributed/_composable/fsdp/test_fully_shard_ignore_params.py

```diff
@@ -0,0 +1,326 @@
+        result = self.module_b(x) + self.addend
+        return result.sum()
+
+
+class Y(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        p = torch.randn(10, device="cuda")
+        self.p = nn.Parameter(p)
+
+
+class X(nn.Module):
+    def __init__(self) -> None:
+        super().__init__()
+        q = torch.randn(10, device="cuda")
+        self.q = nn.Parameter(q)
+        self.y = Y()
+
+
+def _append_prefix(prefix: str, name: str) -> str:
+    if prefix != "" and name != "":
+        return prefix + "." + name
+    else:
+        return prefix + name
+
+
+def _generate_model_and_input() -> nn.Module:
+    dim = 8
+
+    torch.manual_seed(42)
+    addend = torch.randn((dim, dim), device="cuda")
+
+    torch.manual_seed(70)
+    subend = torch.randn((dim, dim), device="cuda")
+
+    model = A(dim, addend, subend).cuda()
+
+    torch.manual_seed(84)
+    inp = torch.randn((dim, dim), device="cuda")
+
+    return model, inp
+
+
+def _find_name_param_mappings(module: torch.nn.Module, prefix: str):
+    name_to_param_map = {}
+    param_to_name_map = {}
+    for name, param in module.named_parameters(prefix):
+        name_to_param_map[name] = param
+        param_to_name_map[param] = name
+    return name_to_param_map, param_to_name_map
+
+
+def _discover_ddp_ignored_params(module: torch.nn.Module, prefix: str):
+    ddp_ignore_parameters: List[str] = []
+    if isinstance(module, FSDP2):
+        ddp_ignore_parameters =  [
+            name for name, _ in module.named_parameters(prefix)
+        ]
+    else:
+        for name, child in list(module.named_children()):
+            # post order traversal
+            path = _append_prefix(prefix, name)
+            ignored_params = _discover_ddp_ignored_params(child, path)
+            ddp_ignore_parameters.extend(ignored_params)
+
+    return ddp_ignore_parameters
+
+
+def _modify_ddp_ignored_params(ddp_ignored_param_names: list[str],
+                               fsdp_ignored_params: set[torch.nn.Parameter],
+                               name_to_param_map: dict):
+    modified_list = []
+    for name in ddp_ignored_param_names:
+        assert name in name_to_param_map
+        param = name_to_param_map[name]
+        if param not in fsdp_ignored_params:
+            # DDP can ignore only if it is not ignored by FSDP
+            modified_list.append(name)
+    return modified_list
+
+
+def _print_all_params(title: str, name_to_param_map: dict):
+    msg = f"{title}:\n"
+    for name, param in sorted(name_to_param_map.items()):
+        msg += f"name={name}, param={param}\n"
+    print(msg)
+
+
+def _print_param_set(title: str, param_set: set[torch.nn.Parameter], param_to_name_map):
+    msg = f">>> [rank {dist.get_rank()}]: {title}"
+    msg += "{"
+    for param in param_set:
+        assert param in param_to_name_map
+        name = param_to_name_map[param]
+        msg += f"{name}, "
+    msg += "}"
+    print(msg)
```

Comment:
nit: Are these for debugging? Do you want to keep them in the unit test when landgin?